src/Symfony/Component/DependencyInjection/Compiler/RegisterServiceSubscribersPass.php

```diff
@@ -56,14 +56,13 @@ protected function processValue($value, $isRoot = false)
         }
         $class = $value->getClass();
 
-        if (!is_subclass_of($class, ServiceSubscriberInterface::class)) {
-            if (!class_exists($class, false)) {
-                throw new InvalidArgumentException(sprintf('Class "%s" used for service "%s" cannot be found.', $class, $this->currentId));
-            }
-
+        if (!$r = $this->container->getReflectionClass($class)) {
+            throw new InvalidArgumentException(sprintf('Class "%s" used for service "%s" cannot be found.', $class, $this->currentId));
+        } elseif (!$r->isSubclassOf(ServiceSubscriberInterface::class)) {
             throw new InvalidArgumentException(sprintf('Service "%s" must implement interface "%s".', $this->currentId, ServiceSubscriberInterface::class));
         }
-        $this->container->addObjectResource($class);
+        $class = $r->name;
```

Comment:
why is it needed?

Reply:
because the class can be a parameter name that is resolved by getReflectionClass, so we need the resolved name

Reply:
thanks